Temporary revert of r2773.

--- a/cmd-set-option.c
+++ b/cmd-set-option.c
@@ -87,7 +87,6 @@
 	struct winlink				*wl;
 	struct client				*c;
 	struct options				*oo;
-	struct window				*w;
 	const char				*optstr, *valstr;
 	u_int					 i;
 
@@ -148,18 +147,6 @@
 			return (-1);
 	}
 
-	/* Start or stop timers when automatic-rename changed. */
-	if (strcmp (oe->name, "automatic-rename") == 0) {
-		for (i = 0; i < ARRAY_LENGTH(&windows); i++) {
-			if ((w = ARRAY_ITEM(&windows, i)) == NULL)
-				continue;
-			if (options_get_number(&w->options, "automatic-rename"))
-				queue_window_name(w);
-			else if (event_initialized(&w->name_timer))
-				evtimer_del(&w->name_timer);
-		}
-	}
-
 	/* Update sizes and redraw. May not need it but meh. */
 	recalculate_sizes();
 	for (i = 0; i < ARRAY_LENGTH(&clients); i++) {
--- a/names.c
+++ b/names.c
@@ -49,7 +49,9 @@
 	struct window	*w = data;
 	char		*name, *wname;
 
-	queue_window_name(w); /* stopped when option turned off */
+	queue_window_name(w);	/* XXX even if the option is off? */
+	if (!options_get_number(&w->options, "automatic-rename"))
+		return;
 
 	if (w->active->screen != &w->active->base)
 		name = NULL;
--- a/window.c
+++ b/window.c
@@ -294,9 +294,9 @@
 	w->sx = sx;
 	w->sy = sy;
 
+	queue_window_name(w);
+
 	options_init(&w->options, &global_w_options);
-	if (options_get_number(&w->options, "automatic-rename"))
-		queue_window_name(w);
 
 	for (i = 0; i < ARRAY_LENGTH(&windows); i++) {
 		if (ARRAY_ITEM(&windows, i) == NULL) {
